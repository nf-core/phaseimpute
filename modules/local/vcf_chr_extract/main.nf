process VCF_CHR_EXTRACT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bcftools:1.18--h8b25389_0':
        'biocontainers/bcftools:1.18--h8b25389_0' }"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.txt"), emit: chr
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    bcftools \\
        query \\
        -f '%CHROM\\n' \\
        $input \\
        \| uniq \\
        > ${prefix}.txt


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bcftools: \$( bcftools --version |& sed '1!d; s/^.*bcftools //' )
        grep: \$( grep --version |& grep -o -E '[0-9]+\\.[0-9]+' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bcftools: \$( bcftools --version |& sed '1!d; s/^.*bcftools //' )
        uniq: \$( uniq --version |& grep -o -E '[0-9]+\\.[0-9]+' )
    END_VERSIONS
    """
}
