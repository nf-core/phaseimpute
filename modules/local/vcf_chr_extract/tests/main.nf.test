nextflow_process {

    name "Test Process VCF_CHR_EXTRACT"
    script "../main.nf"
    process "VCF_CHR_EXTRACT"

    tag "modules"
    tag "modules_local"
    tag "vcf_chr_extract"

    test("Extract chr from vcf") {
        when {
            process {
                """
                input[0] = Channel.fromList([
                    [
                        [id:'test_bcf'], // meta map
                        file(params.test_data['sarscov2']['illumina']['test_bcf'], checkIfExists: true)
                    ],
                    [
                        [id: 'test2_vcf'], // meta map
                        file(params.test_data['sarscov2']['illumina']['test2_vcf'], checkIfExists: true)
                    ],
                    [
                        [id: 'test2_haplotc_vcf_gz'], // meta map
                        file(params.test_data['homo_sapiens']['illumina']['test2_haplotc_vcf_gz'], checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(process.out.chr.collect{path(it[1]).readLines()}).match("chr") }
            )
        }
    }
}
