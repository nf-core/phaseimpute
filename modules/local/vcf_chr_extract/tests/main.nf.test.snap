{
    "Extract chr from vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2_haplotc_vcf_gz"
                        },
                        "test2_haplotc_vcf_gz.txt:md5,e99d7d1051eee43ceab5563c2d09fcee"
                    ],
                    [
                        {
                            "id": "test2_vcf"
                        },
                        "test2_vcf.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ],
                    [
                        {
                            "id": "test_bcf"
                        },
                        "test_bcf.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe",
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe",
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe"
                ],
                "chr": [
                    [
                        {
                            "id": "test2_haplotc_vcf_gz"
                        },
                        "test2_haplotc_vcf_gz.txt:md5,e99d7d1051eee43ceab5563c2d09fcee"
                    ],
                    [
                        {
                            "id": "test2_vcf"
                        },
                        "test2_vcf.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ],
                    [
                        {
                            "id": "test_bcf"
                        },
                        "test_bcf.txt:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe",
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe",
                    "versions.yml:md5,ef0347837e562ce29da2cf994c87ebbe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T14:52:17.414553172"
    },
    "chr": {
        "content": [
            [
                [
                    "chr21"
                ],
                [
                    "MT192765.1"
                ],
                [
                    "MT192765.1"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-17T14:55:53.822382037"
    }
}